docs/arguments.md

```diff
@@ -4,45 +4,67 @@
 
 The `Arguments` class is used by Windsor to pass arguments [down the invocation pipeline](how-dependencies-are-resolved.md). The class is a simple implementation of non-generic `IDictionary`, but it has some useful capabilities.
 
-:information_source: **Custom `IDictionary` is respected:** When you call `container.Resolve` passing your own custom implementation of `IDictionary` Windsor will respect that and not replace it with `Arguments`. It is advised that you use `Arguments` though.
+Resolution arguments are key/value pairs, keyed either by name (`string`) or type (`System.Type`). How the arguments are added to the `Arguments` collections determines how they are bound to dependencies during resolution.
 
 ### Constructors
 
 The class has several constructors:
+```csharp
+// Empty collection
+new Arguments();
 
-#### `namedArgumentsAsAnonymousType`
-
-You can pass named arguments as properties on anonymous type:
+// Initialize from another collection
+new Arguments(new Dictionary<object, object>());
+```
 
+Other constructors form part of the fluent interface of inserting arguments:
 ```csharp
-new Arguments(new { logLevel = LogLevel.High });
+new Arguments("hostname", "localhost");
+new Arguments(typeof(MyClass), new MyClass());
 ```
 
-:information_source: **Named arguments are not case sensitive:** You can specify `logLevel`, `LogLevel` or even `LOGLEVEL` as property name. All named arguments are matched in case insensitive manner so in all cases the behavior will be the same.
+### Fluent Interface
 
-#### Array of `typedArguments`
+:information_source: **Inserts overwrites:** Insert methods overwrite existing keys rather than throw.
```

Comment:
'Insert overwrites' or 'inserts overwrite'?
Not sure but I think I'd normally say, "rather than throwing."

Reply:
"They overwrite rather than throw" still sounds more awkward to me than `"They overwrite rather than throwing," but a [search](https://www.google.com/search?q=rather+than+ing) is showing me that most people prefer it that way you have it it here, so you should probably disregard. :+1: